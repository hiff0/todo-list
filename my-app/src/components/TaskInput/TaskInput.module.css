.taskinput {
    width: 100%;
    display: flex;
    justify-content: space-between;
    margin: 10px 0;
}

.input {
    width: 100%;
    height: 20px;
    margin-right: 10px;
    border: 2px solid #333;
    border-radius: 5px;
}

.buttons {
    display: flex;
    justify-content: space-between;
    align-items: center;
}