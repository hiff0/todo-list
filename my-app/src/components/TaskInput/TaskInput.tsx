import React, { useState } from "react";
import Button from "../Button/Button";
import { TaskType } from "../../utils/types";
import styles from './TaskInput.module.css';

interface TaskPanel {
    addTask: ({ title }: Omit<TaskType, 'isDone' | 'id'>) => void;
}

function TaskInput({ addTask }: TaskPanel) {
    const [todo, setTodo] = useState({
        title: '',
    })

    const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        const { value } = event.target;
        setTodo({ title: value })
    }

    const onClick = () => {
        addTask({ title: todo.title });
        setTodo({ title: '' })
    }

    return (
        <div className={styles.taskinput}>
            <input type="text" placeholder="Введите задачу" onChange={onChange} value={todo.title} className={styles.input} />
            <Button title="Добавить" onClick={onClick} />
        </div>
    )
}

export default TaskInput;