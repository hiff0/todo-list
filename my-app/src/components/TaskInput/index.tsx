import { useState } from "react";
import Button from "../Button";
import styles from './TaskInput.module.css';
import { useTaskContext } from '../../provider';

interface TaskInputProps {
    mode: 'Добавить' | 'Изменить'
}

function TaskInput({ mode }: TaskInputProps) {
    const { tasks, taskIdForEdit, addTask, changeTask, resetTaskIdForEdit } = useTaskContext();
    const [todo, setTodo] = useState({
        title: '',
    });
    const isEdit = mode === 'Изменить';
    const placeholder = mode === 'Изменить' ? 'Измените задачу' : ' ';


    const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        const { value } = event.target;
        setTodo({ title: value })
    }

    const preventDefault = () => {
        const formButton = document.querySelector('form');
        formButton?.addEventListener('click', (evt) => {
            evt.preventDefault();
        })
    }

    const onClick = () => {
        if (isEdit) {
            return (
                changeTask({ title: todo.title })
            )
        }
        addTask({ title: todo.title });
        setTodo({ title: '' });
    }

    return (
        <form className={styles.taskinput} onClick={preventDefault}>
            <input type="text" placeholder={placeholder} onChange={onChange} value={todo.title} className={styles.input} />
            <div className={styles.buttons}>
                <Button title={mode} onClick={onClick} />
                {
                    isEdit && <Button title="Отменить" onClick={resetTaskIdForEdit} />
                }
            </div>
        </form>
    )
}

export default TaskInput;