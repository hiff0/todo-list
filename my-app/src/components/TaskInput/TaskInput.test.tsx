import { render, screen } from '@testing-library/react';
import TaskInput from './TaskInput';

describe('TaskInput component', () => {
    it('renders', () => {
        render(<TaskInput mode='Добавить' />);
        expect(screen.getByDisplayValue('')).toBeInTheDocument();
    });

    it('renders with edit mode', () => {
        render(<TaskInput mode='Изменить' />);
        expect(screen.getByText('Отменить')).toBeInTheDocument();
        expect(screen.getByText('Изменить')).toBeInTheDocument();
        expect(screen.queryByText('Добавить')).toBeNull();
    })

    it('renders with add mode', () => {
        render(<TaskInput mode='Добавить' />);
        expect(screen.getByText('Добавить')).toBeInTheDocument();
        expect(screen.queryByText('Изменить')).toBeNull();
    })

    it('snapshot with edit mode', () => {
        const taskInput = render(<TaskInput mode='Изменить' />);
        expect(taskInput).toMatchSnapshot();
    })

    it('snapshot with add mode', () => {
        const taskInput = render(<TaskInput mode='Добавить' />);
        expect(taskInput).toMatchSnapshot();
    })
})