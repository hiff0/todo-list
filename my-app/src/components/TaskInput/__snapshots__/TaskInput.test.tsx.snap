// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskInput component snapshot with add mode 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="taskinput"
      >
        <input
          class="input"
          placeholder=" "
          type="text"
          value=""
        />
        <div
          class="buttons"
        >
          <button
            style="margin: 0px; border-radius: 5px; cursor: pointer; height: 30px; background: rgb(101, 106, 247);"
          >
            Добавить
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="taskinput"
    >
      <input
        class="input"
        placeholder=" "
        type="text"
        value=""
      />
      <div
        class="buttons"
      >
        <button
          style="margin: 0px; border-radius: 5px; cursor: pointer; height: 30px; background: rgb(101, 106, 247);"
        >
          Добавить
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TaskInput component snapshot with edit mode 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="taskinput"
      >
        <input
          class="input"
          placeholder="Измените задачу"
          type="text"
          value=""
        />
        <div
          class="buttons"
        >
          <button
            style="margin: 0px; border-radius: 5px; cursor: pointer; height: 30px; background: rgb(214, 126, 126);"
          >
            Изменить
          </button>
          <button
            style="margin: 0px; border-radius: 5px; cursor: pointer; height: 30px; background: rgb(240, 14, 14);"
          >
            Отменить
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="taskinput"
    >
      <input
        class="input"
        placeholder="Измените задачу"
        type="text"
        value=""
      />
      <div
        class="buttons"
      >
        <button
          style="margin: 0px; border-radius: 5px; cursor: pointer; height: 30px; background: rgb(214, 126, 126);"
        >
          Изменить
        </button>
        <button
          style="margin: 0px; border-radius: 5px; cursor: pointer; height: 30px; background: rgb(240, 14, 14);"
        >
          Отменить
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
